### PR TITLE
Fine tune GRF magnitudes per trial

### PR DESCRIPTION
**Pull request by keenon**
This adds support for fine tuning the magnitude of GRF forces per trial. The idea is that across multiple trials, if you have different step frequency, you can get slightly different average vertical force over time (and consequently different average mass). I've added support to the linear fitting step to do very fine adjustments to the magnitude of the GRF force data per trial, so that it can (hopefully) compensate. Local testing against Sam's data looks promising, but I probably have a different setup than you, so I'd like your review!

Sadly, there's a bunch of random line-wrapping changes here also because my autoformatter must be on slightly different settings :/

### CODE REVIEW

File: dart/biomechanics/DynamicsFitter.cpp [hunk 1]
```diff
@@ -10443,11 +10440,11 @@ void DynamicsFitter::zeroLinearResidualsOnCOMTrajectory(
...
       // coordinates of COM over time, given those inputs. Additionally, there
       // are columns to account for small force plate rotation errors (3 angles
-      // per force plate) and for instantaneous COM velocities at time steps with
-      // missing ground reaction force data.
+      // per force plate) and for instantaneous COM velocities at time steps
+      // with missing ground reaction force data.
       Eigen::MatrixXs trialLinearMapToPositions = Eigen::MatrixXs::Zero(
-          numTimesteps * 3 + (numForcePlates * 3) + (numMissingSteps * 3),
-          6 + (numForcePlates * 3) + (numMissingSteps * 3) + 1);
+          numTimesteps * 3 + (numForcePlates * 3) + (numMissingSteps * 3) + 1,
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10446]: Here I add a column for the new "scaleGRF" value, and I add a row to regularize it.

File: dart/biomechanics/DynamicsFitter.cpp [hunk 2]
```diff
@@ -10528,9 +10529,9 @@ void DynamicsFitter::zeroLinearResidualsOnCOMTrajectory(
           int timestepsSinceDv = t - missingIndex;
           int mapping = missingStepMappings[j];
 
-          int missingXCol = 6 + mapping * 3;
-          int missingYCol = 6 + mapping * 3 + 1;
-          int missingZCol = 6 + mapping * 3 + 2;
+          int missingXCol = 6 + (numForcePlates * 3) + mapping * 3;
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10532]: This was just a straightforward bug in the previous version, unrelated to the current fix.

File: dart/biomechanics/DynamicsFitter.cpp [hunk 3]
```diff
@@ -10628,6 +10629,10 @@ void DynamicsFitter::zeroLinearResidualsOnCOMTrajectory(
             = regularizeUnobservedTimesteps;
       }
 
+      int scaleGRFRow
+          = (numTimesteps * 3) + (numForcePlates * 3) + (numMissingSteps * 3);
+      trialLinearMapToPositions(scaleGRFRow, scaleGRFCol) = regularizeScaleGRF;
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10634]: Add the regularization value to the matrix, which just directly multiplies the `scaleGRF` on the input and tries to make it match 0 as part of the least squares fit.

File: javascript/webpack.config.devserver.js [hunk 4]
```diff
@@ -47,7 +47,13 @@ module.exports = {
     })
   ],
   devServer: {
-    contentBase: path.join(__dirname, "dist"),
+    static: path.join(__dirname, "dist"),
```
Comment (nickbianco) [inline javascript/webpack.config.devserver.js:50]: Did you intend to have this change in this PR?

File: unittests/unit/test_DynamicsFitter.cpp [hunk 5]
```diff
@@ -3499,7 +3505,8 @@ std::pair<std::vector<MarkerInitialization>, OpenSimFile> runMarkerFitter(
   MarkerFitter fitter(standard.skeleton, standard.markersMap);
   fitter.setInitialIKSatisfactoryLoss(0.005);
   fitter.setInitialIKMaxRestarts(50);
-  fitter.setIterationLimit(500);
+  // TODO: revert me back up to 500
+  fitter.setIterationLimit(200);
```
Comment (nickbianco) [inline unittests/unit/test_DynamicsFitter.cpp:3509]: Revert back up to 500?

**Review by nickbianco — CHANGES_REQUESTED** [review]
Just two small things to address.

I just ran these changes locally and it worked great!

File: unittests/unit/test_DynamicsFitter.cpp (hunk 5 above)
Comment (keenon) [reply unittests/unit/test_DynamicsFitter.cpp]: Excellent catch, this is now fixed :)

File: javascript/webpack.config.devserver.js (hunk 4 above)
Comment (keenon) [reply javascript/webpack.config.devserver.js:50]: I had to fix this in order to get the JS visualizer to work (I guess Webpack silently changed in non-backward-compatible ways... >:( JS people...) and I was using the visualizer to debug. I guess if we were really being grown ups it'd be a separate PR, but that's a hassle so let's merge it in this one, if that's ok :)

Comment (nickbianco) [reply javascript/webpack.config.devserver.js:50]: Totally fine with me, just checking!

**Review by nickbianco — APPROVED** [review]
LGTM!

**Pull request merged**